plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -568,6 +577,11 @@ instance CaseBuiltin DefaultUni where
                 []       -> Right $ HeadOnly $ branches Vector.! 1
                 (y : ys) -> Right $ headSpine (branches Vector.! 0) [someValueOf ty y, someValueOf uni ys]
             | otherwise            -> Left $ outOfBoundsErr someVal branches
```

Comment:
Hm, I thought we've discussed it, but perhaps it was something else. We should have a `CaseResult` data type that is properly strict and removes the `Either` indirection that we have here.

It'll mean adding a constructor for errors to `HeadSpine`, since we only use that for casing on constants nowadays anyway.

Reply:
Will make a separate optimization PR